memgpt/memory.py

```diff
@@ -16,96 +18,214 @@
     validate_date_format,
 )
 
-# from llama_index import Document
-# from llama_index.node_parser import SimpleNodeParser
 
+class MemoryModule(BaseModel):

```

Comment:
Looking at this section here I'm not entirely clear on what a `MemoryModule` is vs a `BaseMemory`, and by association, I'm not clear on what the attributes represent (so it's hard to follow along). 

More unique names, descriptive class docstrings, and `Field` annotations for attributes with descriptions would help. 

For example, is the `limit` attribute on `MemoryModule` the number of characters in a `Memory`, or the number of `Memory` objects, or something else?

`value` is another one - it almost looks like that's another subschema here. I'm not sure what values are compared to memories.

Reply:
The `limit` is how many characters can be in the string representation of `value`, since the string representation of `value` is placed into the context window for the LLM. 

We are actually planning to eventually store the `MemoryModule` (or whatever its renamed to) as a DB row that is read from at inference time (when the context is "compiled"). This would allow multiple agents to share a given row (e.g. where `value` represents organizational memory across agents). So maybe another potential name for the class could be `StateVariable` or `ContextVariable`.